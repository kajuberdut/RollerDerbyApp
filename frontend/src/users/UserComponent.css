.UserComponent {
    background-color: white;
    margin-top: 15px;
    height: 50px;
    width: 70%;
    margin-left: 12%;
    box-shadow: 0 2px 2px 0 rgba(0,0,0,0.14), 0 3px 1px -2px rgba(0,0,0,0.12), 0 1px 5px 0 rgba(0,0,0,0.2);
}

.UserComponent-Title h4:hover,
.UserComponent-Title h4[data-hover] {
  color: black; 
}