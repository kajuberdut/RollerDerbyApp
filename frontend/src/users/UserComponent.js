import React from "react";
import { NavLink } from "react-router-dom";
import "./UserComponent.css";
import { useParams } from "react-router-dom";
import FastApi from "../Api";

  /**  
  * Card component for users
  */

  function UserComponent({indUser, getPendingInvites}) {

    const pathname = window.location.pathname;
    const teamId = useParams(); 

    /** Retrieve user from local storage*/
  
    const user = JSON.parse(localStorage.getItem('user'));
    
    /** API get request for teams */

    async function sendInvite() {

      try {
        let data = {
          inviteId: 0,
          teamId: teamId.id,
          senderId: user.userId, 
          recipientId: indUser.userId,
          status: 'pending'
        }

        await FastApi.addTeamInvite(data);
        
      } catch (errors) {

        return { success: false, errors };
      }
    }

    /** Handle click of button  */

    async function handleClick(e) {
      e.preventDefault(); 
      sendInvite(); 
      getPendingInvites(); 
    }

     /** Render the card component */
      
      return (
      <div key={"UserComponent-" + indUser.userId } className="UserComponent" style={{width: '77%', height: 'auto', minHeight: '90px', borderRadius: '5px', overflow: 'hidden'}}> 
       <NavLink to={`/users/${indUser.userId}`} className="UserComponent-Link" style={{color: '#555555', textDecoration: 'none'}}>
        <div style={{ display: 'flex'}}>
            <div style={{paddingTop: '8px', paddingLeft: '20px', paddingRight: '20px'}} >
              {/* { indUser.image && <img src={indUser.image}
                alt="Skater placeholder image" className="img-thumbnail" style={{ width: '60px', borderRadius: '50%', backgroundColor: '#d1d2d4', border: '2px solid black', padding: '0px'}}/> 
              } */}
            </div>           
            <h4 style={{paddingTop: '30px'}}>{indUser.username}</h4>
            {pathname !== "/users" && <div style={{paddingLeft: '60%', paddingTop: '7%',position: 'absolute'}}> <button onClick={handleClick} style={{borderRadius: '5px', height: '33px', top: '10px', right: '20px' }}>Invite</button> </div>}
        </div>
        </NavLink>
      </div>
        );
  }
  
  export default UserComponent
