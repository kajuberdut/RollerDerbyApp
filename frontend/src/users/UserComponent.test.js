import { render, screen, waitFor } from '@testing-library/react';
import UserComponent from './UserComponent.js';
import { BrowserRouter } from "react-router-dom";
import FastApi from '../Api'; 

// * Passing as of 2/14/24
// npm test UserComponent.test.js
// must be in frontend directory 

// Mock FastApi methods
jest.mock('../Api', () => ({
    addTeamInvite: jest.fn()
  }));

// Mock useParams
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'), 
    useParams: jest.fn(),
  }));

describe("checks rendering of card component", () => {  

    let indUser = {
        userId: 2, 
        image: "afjakjdfkjkl",
        username: "User2"
    }

    test('renders card component form page', () => {

        render(
            <BrowserRouter>
                    <UserComponent indUser={indUser} />
            </BrowserRouter>);
        });


    test('matches snapshot', function() {
        const { asFragment } = render(
            <BrowserRouter>
                    <UserComponent indUser={indUser} />
            </BrowserRouter>
        );
        expect(asFragment()).toMatchSnapshot(); 
    })


});

describe("asserts user component text", () => {  

    let indUser = {
        userId: 2, 
        image: "afjakjdfkjkl",
        username: "User2"
    }

    const mockAddTeamInvite = {
      };

      beforeEach(() => {
        FastApi.addTeamInvite.mockResolvedValue(mockAddTeamInvite);
      });
    
      afterEach(() => {
        jest.clearAllMocks();
      });
    

    test('renders card details on page', async () => {

        render(
            <BrowserRouter>
                    <UserComponent indUser={indUser} />
            </BrowserRouter>
        )

        await waitFor(() => {
            expect(screen.getByText('User2')).toBeInTheDocument();
            expect(screen.queryByText('Invite')).toBeInTheDocument();
            
        })

    });

});