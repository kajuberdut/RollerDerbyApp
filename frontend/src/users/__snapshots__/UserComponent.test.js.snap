// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks rendering of card component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="UserComponent"
    style="width: 77%; height: auto; min-height: 100px; border-radius: 5px; overflow: hidden;"
  >
    <a
      class="UserComponent-Link"
      href="/users/2"
      style="color: rgb(85, 85, 85); text-decoration: none;"
    >
      <div
        style="display: flex;"
      >
        <div
          style="padding-top: 8px; padding-left: 20px; padding-right: 0px;"
        />
        <h4
          style="padding-top: 30px; padding-right: 20px; overflow: hidden; max-width: 70%; padding-bottom: 5px;"
        >
          User2
        </h4>
        <div
          style="padding-left: 60%; padding-top: 40px; position: absolute;"
        >
           
          <button
            style="border-radius: 5px; height: 33px; top: 10px; right: 20px;"
          >
            Invite
          </button>
           
        </div>
      </div>
    </a>
  </div>
</DocumentFragment>
`;
